pkg/storage/mvcc.go

```diff
@@ -3731,7 +3737,7 @@ func MVCCGarbageCollect(
 	meta := &enginepb.MVCCMetadata{}
 	for _, gcKey := range keys {
 		encKey := MakeMVCCMetadataKey(gcKey.Key)
-		ok, metaKeySize, metaValSize, err := mvccGetMetadata(iter, encKey, meta)
+		ok, metaKeySize, metaValSize, hiddenByRangeDel, err := mvccGetMetadata(iter, encKey, meta)
```

Comment:
`hiddenByRangeDel` is a misnomer. The returned point meta isn't hidden by an MVCC range tombstone, it's a synthetic point tombstone that's hiding the points below it. Consider e.g. `synthetic`. But we may not even need it, see separate comment on handling removal of all point keys.